tests/jballerina-unit-test/src/test/resources/test-src/expressions/binaryoperations/equal_and_not_equal_operation.bal

```diff
@@ -1590,13 +1695,39 @@ function testExactEqualityWithUnionOfNonSimpleTypes() {
     test:assertFalse(x2 === h1);
 }
 
-function assert(anydata actual, anydata expected) {
-    if (expected != actual) {
-        typedesc<anydata> expT = typeof expected;
-        typedesc<anydata> actT = typeof actual;
-        string reason = "expected [" + expected.toString() + "] of type [" + expT.toString()
-                            + "], but found [" + actual.toString() + "] of type [" + actT.toString() + "]";
-        error e = error(reason);
-        panic e;
-    }
+function testEqualityByteWithIntSubTypes() {
+    byte a = 120;
+    int b = 120;
+    int:Unsigned8 c = 120;
+    int:Unsigned16 d = 120;
+    int:Unsigned32 e = 120;
+    int:Signed8 f = 120;
+    int:Signed16 g = 120;
+    int:Signed32 h = 120;
+
+    // == equals check
+    test:assertTrue((a == b) && (a == c) && (a == d) && (a == e) &&
+    (a == f) && (a == g) && (a == h));
+    test:assertTrue((b == a) && (c == a) && (d == a) && (e == a) &&
+    (f == a) && (g == a) && (h == a));
+
+    // != equals check
+    test:assertFalse((a != b) && (a != c) && (a != d) && (a != e) &&
+    (a != f) && (a != g) && (a != h));
+    test:assertFalse((b != a) && (c != a) && (d != a) && (e != a) &&
+    (f != a) && (g != a) && (h != a));
```

Comment:
I think, to make this assertion strong we need to use `||` instead of `&&`, otherwise even only if the first parenthesis gives 'false' the whole one will be `false`.